src/gtk/toga_gtk/widgets/internal/rows/texticon.py

```diff
@@ -0,0 +1,74 @@
+import html
+
+from toga_gtk.libs import Gtk, Pango
+from .base import HiddenButtonsRow
+
+
+class TextIconRow(HiddenButtonsRow):
+    """
+    Create a TextIconRow from a toga.sources.Row.
+    A reference to the original row is kept in self.toga_row, this is useful for comparisons.
+    """
+    def __init__(self, factory: callable, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # This is the factory of the DetailedList implementation.
+        self.factory = factory
+
+        self.icon = self.get_icon(self.interface, self.factory)
+
+        text = Gtk.Label(xalign=0)
+
+        # The three line below are necessary for right to left text.
+        text.set_hexpand(True)
+        text.set_ellipsize(Pango.EllipsizeMode.END)
+        text.set_margin_end(12)
+
+        text_markup = self.markup(self.interface)
+        text.set_markup(text_markup)
+
+        content = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL)
+        vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
+
+        vbox.pack_start(text, True, True, 0)
+
+        if self.icon is not None:
+            content.pack_start(self.icon, False, False, 6)
+
+        content.pack_start(vbox, True, True, 5)
+
+        self.add_content(content)
+
+        self._delete_button = Gtk.Button.new_from_icon_name("user-trash-symbolic", Gtk.IconSize.BUTTON)
+        self._delete_button.connect("clicked", lambda w: self._dl.delete_row(self))
+        self.add_button(self._delete_button)
+
+    @property
+    def title(self):
+        return self.interface.title
+
+    @property
+    def subtitle(self):
+        return self.interface.subtitle
+
+    def get_icon(self, row, factory):
+        if getattr(row, "icon") is None:
+            return None
+        else:
+            row.icon.bind(factory)
+            # TODO: see get_scale_factor() to choose 72 px on hidpi
```

Comment:
Flagging this TODO to be TODONE :-)

Reply:
I did as it's done [here](https://github.com/beeware/toga/blob/19e89a77f1bacddc09110fd5c240ea4ecbf85b2a/src/gtk/toga_gtk/widgets/imageview.py#L30) but I don't have a hidpi screen to test.